app/code/Magento/ConfigurableProduct/Pricing/Price/ConfigurableRegularPrice.php

```diff
@@ -196,26 +196,25 @@ public function _resetState(): void
     }
 
     /**
-     * Check whether Configurable Product have more than one children products
+     * Check whether Configurable options do not have price difference
      *
      * @param SaleableInterface $product
      * @return bool
      */
     public function isChildProductsOfEqualPrices(SaleableInterface $product): bool
```

Comment:
Should an early return be added to the beginning of this method to exit if the product is not configurable?

```php
    if ($product->getTypeId() !== 'configurable') {
        return false;
    }
```

Reply:
Thank you for your comment!
I've just realized that we don't need any arguments, as an object of this class will already have a product assigned.